Bucketeer/Sources/Internal/Evaluation/EvaluationStorageImpl.swift

```diff
@@ -0,0 +1,129 @@
+final class EvaluationStorageImpl: EvaluationStorage {
+
+    var currentEvaluationsId: String {
+        get {
+            return evaluationUserDefaultsDao.currentEvaluationsId
+        }
+        set {
+            evaluationUserDefaultsDao.setCurrentEvaluationsId(value: newValue)
+        }
+    }
+
+    var featureTag: String {
+        get {
+            return evaluationUserDefaultsDao.featureTag
+        }
+        set {
+            evaluationUserDefaultsDao.setFeatureTag(value: newValue)
+        }
+    }
+
+    var evaluatedAt: String {
+        get {
+            return evaluationUserDefaultsDao.evaluatedAt
+        }
+        set {
+            evaluationUserDefaultsDao.setEvaluatedAt(value: newValue)
+        }
+    }
+
+    var userAttributesUpdated: Bool {
+        get {
+            return evaluationUserDefaultsDao.userAttributesUpdated
+        }
+        set {
+            evaluationUserDefaultsDao.setUserAttributesUpdated(value: newValue)
+        }
+    }
+
+    private let userId: String
+    // Expected SQL Dao
+    private let evaluationDao: EvaluationDao
+    // Expected in-memory cache Dao
+    private let evaluationMemCacheDao: EvaluationMemCacheDao
+    private let evaluationUserDefaultsDao: EvaluationUserDefaultsDao
+
+    init(
+        userId: String,
+        evaluationDao: EvaluationDao,
+        evaluationMemCacheDao: EvaluationMemCacheDao,
+        evaluationUserDefaultsDao: EvaluationUserDefaultsDao
+    ) {
+        self.userId = userId
+        self.evaluationDao = evaluationDao
+        self.evaluationUserDefaultsDao = evaluationUserDefaultsDao
+        self.evaluationMemCacheDao = evaluationMemCacheDao
+        try? refreshCache()
+    }
+
+    func get(userId: String) throws -> [Evaluation] {
+        evaluationMemCacheDao.get(key: userId) ?? []
+    }
+
+    func deleteAllAndInsert(userId: String, evaluations: [Evaluation], evaluatedAt: String) throws {
+        try evaluationDao.startTransaction {
+            try evaluationDao.deleteAll(userId: userId)
+            try evaluationDao.put(userId: userId, evaluations: evaluations)
+        }
+        // Update cache directly after we have force update
+        evaluationMemCacheDao.set(key: userId, value: evaluations)
+        self.evaluatedAt = evaluatedAt
+    }
+
+    func update(evaluations: [Evaluation], archivedFeatureIds: [String], evaluatedAt: String) throws -> Bool {
+        var archivedEvaluations = [Evaluation]()
```

Comment:
This can be deleted since it is not being used anymore.

Reply:
we are still using this method. So I'd like to keep it.
```swift
func update(evaluations: [Evaluation], archivedFeatureIds: [String], evaluatedAt: String) throws -> Bool {
```

Reply:
I meant the variable `var archivedEvaluations = [Evaluation]()` is not being used, not the function.

Reply:
I removed this